File: src/main/java/ru/mm/surv/codecs/webm/incubator/streamm/ControlledStream.java
package ru.mm.surv.codecs.webm.incubator.streamm;

public class ControlledStream extends Stream {

    private final int maxClients;
    private int numClients;

    public ControlledStream(int maxClients) {
        this.maxClients = maxClients;
    }

    public boolean subscribe(boolean force) {
        if (force || numClients < maxClients) {
            numClients++;
            refresStatus();
            return true;
        }
        return false;
    }

    public void unsubscribe() {
        numClients--;
        refresStatus();
    }

    protected void refresStatus() {
        ServerStatusEvent event = new ServerStatusEvent();
        event.setClientCount(numClients);
        postEvent(event);
    }
}

File: src/main/kotlin/ru/mm/surv/codecs/webm/incubator/streamm/ControlledStream.kt
package ru.mm.surv.codecs.webm.incubator.streamm

class ControlledStream(private val maxClients: Int) : Stream() {
    private var numClients = 0

    fun subscribe(force: Boolean): Boolean {
        if (force || numClients < maxClients) {
            numClients++
            refreshStatus()
            return true
        }
        return false
    }

    fun unsubscribe() {
        numClients--
        refreshStatus()
    }

    private fun refreshStatus() {
        val event = ServerStatusEvent()
        event.clientCount = numClients
        postEvent(event)
    }
}